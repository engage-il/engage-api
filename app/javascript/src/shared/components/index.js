// @flow
export { Defer } from './Defer'
export { TranslateAndFade } from './TranslateAndFade'
export { NotificationView } from './NotificationView'

export { Link } from './Link'
export type { LinkProps } from './Link'
export { CopyLink } from './CopyLink'
export { Button } from './Button'
export { CloseButton } from './CloseButton'

export { BillSearch } from './BillSearch'
