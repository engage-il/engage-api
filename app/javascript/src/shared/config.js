// @flow
/* globals $Keys */
type Config = {|
  graphUrl: string,
|}

const configs = {
  development: {
    graphUrl: 'http://localhost:3000/api/graphql'
  },
  staging: {
    graphUrl: 'https://legislated-staging.herokuapp.com/api/graphql'
  },
  production: {
    graphUrl: 'https://legislated.herokuapp.com/api/graphql'
  },
  server: {
    graphUrl: 'http://localhost:3000/api/graphql'
  }
}

type Environment
  = $Keys<typeof configs>

function getEnv (): Environment {
  const env = process.env.ENVIRONMENT

  switch (env) {
    case 'development':
    case 'staging':
    case 'production':
    case 'server':
      return env
    default:
      throw new Error(`Invalid environment specified: ${env || 'null'}`)
  }
}

export function loadConfig () {
  let env = getEnv()
  const config: Config = configs[env]

  return {
    ...config,
    env
  }
}

export default loadConfig()
