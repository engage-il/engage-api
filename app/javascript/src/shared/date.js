export { default as addDays } from 'date-fns/add_days'
export { default as addHours } from 'date-fns/add_hours'
export { default as addMonths } from 'date-fns/add_months'
export { default as differenceInHours } from 'date-fns/difference_in_hours'
export { default as endOfDay } from 'date-fns/end_of_day'
export { default as formatDate } from 'date-fns/format'
export { default as parseDate } from 'date-fns/parse'
export { default as startOfDay } from 'date-fns/start_of_day'