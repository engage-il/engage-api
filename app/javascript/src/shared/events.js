// @flow
import EventEmitter from 'eventemitter3'

class Events extends EventEmitter {
  setAuthHeader = '@@events/set-auth-header'
  setEnvironment = '@@events/set-environment'
}

export const events = new Events()
