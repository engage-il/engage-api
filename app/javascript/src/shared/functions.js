// @flow
import { omit } from 'lodash'

export function withoutRouter (props: Object) {
  return omit(props, [
    'location',
    'history',
    'match',
    'staticContext'
  ])
}

export function sleep (duration: number): Promise<void> {
  return new Promise(resolve => setTimeout(resolve, duration))
}
