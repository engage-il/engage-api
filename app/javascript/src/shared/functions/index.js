// @flow
export {
  isSignedIn,
  signIn,
  signOut
} from './auth'

export {
  now,
  sleep,
  alpha,
  removeRouterProps
} from './utils'
