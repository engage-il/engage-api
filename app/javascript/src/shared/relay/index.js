// @flow
export { currentEnvironment } from './currentEnvironment'
export { cacheResolvers } from './createQuery'
export { createPaginationCacheResolver } from './createPaginationCacheResolver'
