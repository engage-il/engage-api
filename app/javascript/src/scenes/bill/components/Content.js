// @flow
import React, { Component } from 'react'
import { createFragmentContainer, graphql } from 'react-relay'
import { withRouter } from 'react-router-dom'
import type { ContextRouter } from 'react-router-dom'
import { Actions } from './Actions'
import { Element } from './Element'
import { CopyLink } from 'shared/components'
import { stylesheet, colors, mixins } from 'shared/styles'
import { now, differenceInHours, formatDate, parseDate } from 'shared/date'
import type { Bill } from 'shared/types'

let Content = class Content extends Component {
  props: {
    bill: Bill
  } & ContextRouter

  // lifecycle
  render () {
    const { bill, location } = this.props

    const date = parseDate(bill.hearing.date)
    const hoursLeft = differenceInHours(date, now())
    const formattedDate = formatDate(date, 'M/D/YY h:mm A')

    return <div>
      <div {...rules.header}>
        <section>
          <h1>{bill.title}</h1>
          <h4>{bill.documentNumber}</h4>
        </section>
        <section>
          <CopyLink value={location.pathname} />
        </section>
      </div>
      <div {...rules.body}>
        <div {...rules.column}>
          <Element label='State Synopsis'>{bill.summary}</Element>
        </div>
        <div {...rules.column}>
          <Element label='Hearing Date'>
            <span>{formattedDate}</span>
            {hoursLeft < 24 && hoursLeft > 0 && <span {...rules.hoursLeft}>
              {`(${hoursLeft} hours left)`}
            </span>}
          </Element>
          <Element label='Committee'>{bill.committee.name}</Element>
        </div>
      </div>
      <Actions bill={bill} />
    </div>
  }
}

Content = createFragmentContainer(withRouter(Content), graphql`
  fragment Content_bill on Bill {
    documentNumber
    title
    summary
    sponsorName
    hearing {
      date
    }
    committee {
      name
    }
    chamber {
      name
    }
    ...Actions_bill
  }
`)

const rules = stylesheet({
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    marginBottom: 15,
    '> section': {
      display: 'flex',
      flexDirection: 'column'
    },
    ...mixins.mobile.glam({
      flexDirection: 'column',
      '> section:first-child': {
        marginBottom: 10
      },
      '> h1': {
        marginBottom: 5
      }
    })
  },
  body: {
    ...mixins.borders.low(['bottom']),
    display: 'flex',
    flexBasis: 0,
    paddingBottom: 15,
    marginBottom: 15,
    ...mixins.mobile.glam({
      flexDirection: 'column'
    })
  },
  column: {
    flex: 1,
    ':not(:last-of-type)': {
      marginRight: 15
    },
    '> *:not(:last-child)': {
      marginBottom: 10
    },
    ...mixins.mobile.glam({
      marginBottom: 15,
      ':last-child': {
        marginBottom: 0
      }
    })
  },
  hoursLeft: {
    marginLeft: 5,
    color: colors.secondary
  }
})

export { Content }
