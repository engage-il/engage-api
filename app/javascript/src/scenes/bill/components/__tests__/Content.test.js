/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import { defaultsDeep } from 'lodash'
import { Content } from '../Content'
import { now, addHours } from 'shared/date'

jest.mock('shared/date', () => {
  const actual = require.requireActual('shared/date')
  const date = new Date(2017, 2, 14, 5, 0, 0, 0)

  return {
    ...actual,
    now: () => date
  }
})

// subject
let subject

const defaults = {
  bill: {
    documentNumber: 'HB1234',
    title: 'Foo',
    summary: 'A bill, fantastic',
    hearing: {
      date: addHours(now(), 11.9)
    },
    committee: {
      name: 'Many Pointed Bills'
    }
  },
  location: {
    pathname: '/test-path'
  }
}

function loadSubject (props = {}) {
  subject = shallow(<Content {...defaultsDeep(props, defaults)} />)
}

// specs
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('renders properly', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })

  it('does not show the hours left when over 23 hours away', () => {
    loadSubject({
      bill: {
        hearing: { date: addHours(now(), 24.1) }
      }
    })

    expect(subject).toMatchSnapshot()
  })
})
