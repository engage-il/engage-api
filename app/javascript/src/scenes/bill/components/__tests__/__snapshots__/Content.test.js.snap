// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render does not show the hours left when over 23 hours away 1`] = `
<div>
  <div
    data-css-1wwvkuu=""
  >
    <section>
      <h1>
        Foo
      </h1>
      <h4>
        HB1234
      </h4>
    </section>
    <section>
      <CopyLink
        value="placeholder"
      />
    </section>
  </div>
  <div
    data-css-1bq0tws=""
  >
    <div
      data-css-1qww4bd=""
    >
      <Element
        label="State Synopsis"
      >
        A bill, fantastic
      </Element>
    </div>
    <div
      data-css-1qww4bd=""
    >
      <Element
        label="Hearing Date"
      >
        <span>
          3/15/17 5:36 AM
        </span>
      </Element>
      <Element
        label="Committee"
      >
        Many Pointed Bills
      </Element>
    </div>
  </div>
  <Actions
    bill={
      Object {
        "committee": Object {
          "name": "Many Pointed Bills",
        },
        "documentNumber": "HB1234",
        "hearing": Object {
          "date": 2017-03-15T10:36:35.500Z,
        },
        "summary": "A bill, fantastic",
        "title": "Foo",
      }
    }
  />
</div>
`;

exports[`#render renders properly 1`] = `
<div>
  <div
    data-css-1wwvkuu=""
  >
    <section>
      <h1>
        Foo
      </h1>
      <h4>
        HB1234
      </h4>
    </section>
    <section>
      <CopyLink
        value="placeholder"
      />
    </section>
  </div>
  <div
    data-css-1bq0tws=""
  >
    <div
      data-css-1qww4bd=""
    >
      <Element
        label="State Synopsis"
      >
        A bill, fantastic
      </Element>
    </div>
    <div
      data-css-1qww4bd=""
    >
      <Element
        label="Hearing Date"
      >
        <span>
          3/14/17 5:24 PM
        </span>
        <span
          data-css-1fo47jf=""
        >
          (11 hours left)
        </span>
      </Element>
      <Element
        label="Committee"
      >
        Many Pointed Bills
      </Element>
    </div>
  </div>
  <Actions
    bill={
      Object {
        "committee": Object {
          "name": "Many Pointed Bills",
        },
        "documentNumber": "HB1234",
        "hearing": Object {
          "date": 2017-03-14T22:24:35.500Z,
        },
        "summary": "A bill, fantastic",
        "title": "Foo",
      }
    }
  />
</div>
`;
